tools/matrix-bot/logger.ts

```diff
@@ -22,24 +17,59 @@ export class MessagesLogger {
     if (!eventId) {
       return;
     }
-    if (!msg.includes("graypaper.fluffylabs.dev")) {
+    const gpLinks = findLinks(msg);
+    if (!gpLinks.length) {
       return;
     }
 
+    // if there are several links in the message, find the link with the latest version
+    let versionName: string | undefined;
+
+    if (gpLinks.length === 1) {
+      versionName = parseLink(gpLinks[0], this.meta)?.versionName;
+    } else {
+      versionName = gpLinks
+        .map((link) => parseLink(link, this.meta))
+        .filter(Boolean)
+        .map((link) => {
+          const version = link?.versionName || "";
+          const sortKey = version
+            .split(".")
+            .map((num) => num.padStart(3, "0"))
+            .join("");
+          return { version, sortKey };
+        })
+        .sort((a, b) => a.sortKey.localeCompare(b.sortKey))
+        .pop()?.version;
+    }
```

Comment:
```suggestion
      versionName = gpLinks
        .map((link) => parseLink(link, this.meta))
        .filter(Boolean)
        .map((link) => {
          const version = link?.versionName || "";
          const sortKey = version
            .split(".")
            .map((num) => num.padStart(3, "0"))
            .join("");
          return { version, sortKey };
        })
        .sort((a, b) => a.sortKey.localeCompare(b.sortKey))
        .pop()?.version;
```

I'd drop the special casing for 1-element arrays here.